libcloudforensics/providers/azure/internal/compute.py

```diff
@@ -0,0 +1,275 @@
+          AZDisk object.
+    """
+    disks = self.az_account.ListDisks(
+        resource_group_name=self.resource_group_name)
+    vm_disks = self.az_account.compute_client.virtual_machines.get(
+        self.resource_group_name, self.name).storage_profile
+    vm_disks_names = [disk.name for disk in vm_disks.data_disks]
+    vm_disks_names.append(vm_disks.os_disk.name)
+    return {disk_name: disks[disk_name] for disk_name in vm_disks_names}
+
+
+class AZDisk(AZComputeResource):
+  """Class that represents Azure disks."""
+
+  def __init__(self,
+               az_account: 'account.AZAccount',
+               resource_id: str,
+               name: str,
+               region: str,
+               zones: Optional[List[str]] = None) -> None:
+    """Initialize the AZDisk class.
+
+    Args:
+      az_account (AZAccount): An Azure account object.
+      resource_id (str): The Azure ID of the disk.
+      name (str): The disk name.
+      region (str): The region in which the disk is located.
+      zones (List[str]): Optional. Availability zone within the region where
+          the disk is located.
+    """
+    super(AZDisk, self).__init__(az_account,
+                                 resource_id,
+                                 name,
+                                 region,
+                                 zones=zones)
+
+  def Snapshot(self,
+               snapshot_name: Optional[str] = None,
+               tags: Optional[Dict[str, str]] = None) -> 'AZSnapshot':
+    """Create a snapshot of the disk.
+
+    Args:
+      snapshot_name (str): Optional. A name for the snapshot. If none
+          provided, one will be generated based on the disk's name.
+      tags (Dict[str, str]): Optional. A dictionary of tags to add to the
+          snapshot, for example {'TicketID': 'xxx'}.
+
+    Returns:
+      AZSnapshot: A snapshot object.
+
+    Raises:
+      ValueError: If the snapshot name does not comply with the RegEx.
+      RuntimeError: If the snapshot could not be created.
+    """
+
+    if not snapshot_name:
+      snapshot_name = self.name + '_snapshot'
+      truncate_at = 80 - 1
+      snapshot_name = snapshot_name[:truncate_at]
+      if not common.REGEX_SNAPSHOT_NAME.match(snapshot_name):
+        raise ValueError('Snapshot name {0:s} does not comply with '
+                         '{1:s}'.format(snapshot_name,
+                                        common.REGEX_SNAPSHOT_NAME.pattern))
+
+    creation_data = {
+        'location': self.region,
+        'creation_data': {
+            'sourceResourceId': self.resource_id,
+            'create_option': DiskCreateOption.copy
+        }
+    }
+
+    if tags:
+      creation_data['tags'] = tags
+
+    try:
+      request = self.az_account.compute_client.snapshots.create_or_update(
+          self.resource_group_name,
+          snapshot_name,
+          creation_data)
+      while not request.done():
+        sleep(5)  # Wait 5 seconds before checking snapshot status again
```

Comment:
Same, do we wanna wait forever here?

Reply:
closing this comment (same question as above) and we can discuss on the other one for better visibility